nbs/18_callback.fp16.ipynb

```diff
@@ -202,18 +202,49 @@
     "    def after_loss(self): self.autocast.__exit__(None, None, None)\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #24.            "Pretend to be an optimizer for <code>GradScaler</code>"
</pre><p>Not so sure, but I wonder, if this class delegates things to <code>GradScaler</code>  isnt more like just "Pretend to be an optmizer which has param_groups"... so it is not being like an optimizer for gradScaler, but to whoeveer is calling this property on a <code>MixedPrecision</code>  Instance?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/muellerzr/fastai-docment-sprint/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
The Learner still uses its normal optimizer. We fake out the GradScaler in before_step, specifically this line: `self.scaler.step(self)` where GradScaler expects us to pass in an optimizer.